### PR TITLE
Improve snippet organization in SQL Editor

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

**Review by alaister — COMMENTED** [review]
Sometimes when I move the currently selected snippet (even to the same folder), it gets stuck in a loading state:
Screenshot 2024-07-10 at 10 49 16

**Comment by joshenlim** [comment]
Okay just pushed a fix for the issue you reported :)

File: apps/studio/hooks/ui/useFlag.ts [hunk 1]
```diff
@@ -4,6 +4,9 @@ import { useContext } from 'react'
 export function useFlag<T = any>(name: string) {
   const store: any = useContext(FlagContext)
 
+  // [Joshen] REMOVE THIS AT THE END
+  if (name === 'sqlFolderOrganization') return true
+
```
Comment (alaister) [inline apps/studio/hooks/ui/useFlag.ts:9]: Remove this? 😅
```suggestion
```

**Review by alaister — APPROVED** [review]
Looking good!

I can't seem to use this button to create a snippet?
Screenshot 2024-07-15 at 14 42 22

Also another thing I did notice is snippets don't seem to auto-rename anymore?


Tested so far:
- creating a snippet
- creating a folder
- moving a snippet from root to a folder
- moving a snippet back to root
- moving a snippet from one folder to another

File: apps/studio/hooks/ui/useFlag.ts (hunk 1 above)
Comment (joshenlim) [reply apps/studio/hooks/ui/useFlag.ts:9]: oh crap hahahaha good catch!

**Review by alaister — CHANGES_REQUESTED** [review]
whoops accidentally approved before

**Review by alaister — APPROVED** [review]
LGTM!

Finished testing:
- creating a snippet inside a folder
- renaming a snippet inside a folder
- sharing a snippet in a folder
- unsharing a snippet
- opening in a new tab

**Pull request merged**